storage/engine/engine.go

```diff
@@ -188,6 +188,13 @@ func PutProto(engine Engine, key proto.EncodedKey, msg gogoproto.Message) (keyBy
 // key was not found. On success, returns the length in bytes of the
 // key and the value.
 func GetProto(engine Engine, key proto.EncodedKey, msg gogoproto.Message) (ok bool, keyBytes, valBytes int64, err error) {
+	type protoGetter interface {
```

Comment:
Any interfaces used should generally be public and documented; it can be surprising when you pass an Engine to a function and a method that is not a part of the Engine interface is called. Or we could just add GetProto to the Engine interface to ensure that Batch and Snapshot implement it and pass it along. (there's no good way for wrappers to implement protoGetter iff their wrapped object does).

Instead of a new method on Engine, we may want to change the interface of Engine.Get to take an object to receive the bytes (similar to json.Unmarshaler) instead of returning the bytes. There aren't that many references to Engine.Get so it shouldn't be too hard to change.